src/extensions/mountainview/MVSortingView/MVSortingView.tsx

```diff
@@ -76,33 +76,77 @@ export const openViewsReducer: React.Reducer<View[], OpenViewsAction> = (state:
     else return state
 }
 
-const MVSortingView: FunctionComponent<SortingViewProps> = (props) => {
+const area = (a: 'north' | 'south') => {
+    return a
+}
+
+const MVSortingViewWithCheck: FunctionComponent<SortingViewProps> = (props) => {
+    const {recording, sorting} = props
+
+    const [preprocessingSelection, preprocessingSelectionDispatch] = useReducer(preprocessingSelectionReducer, {filterType: 'none'})
+
+    const preprocessedRecording = useMemo(() => {
+        if (!recording) return recording
+        if (preprocessingSelection.filterType === 'none') {
+            return recording
+        }
+        else if (preprocessingSelection.filterType === 'bandpass_filter') {
+            return {
+                ...recording,
+                recordingObject: {
+                    recording_format: 'filtered',
+                    data: {
+                        filters: [{type: 'bandpass_filter', freq_min: 300, freq_max: 3000, freq_wid: 1000}],
+                        recording: recording.recordingObject
+                    }
+                }
+            }
+        }
+        else {
+            throw Error(`Unexpected filter type: ${preprocessingSelection.filterType}`)
+        }
+    }, [recording, preprocessingSelection])
+
+    return (
+        <PreloadCheck recording={preprocessedRecording} sorting={sorting}>
```

Comment:
The PreloadCheck does the preloading and  passes preloadStatus to the child. The modal dialog is created inside <PreloadCheck />